mlflow/utils/requirements_utils.py

```diff
@@ -672,11 +677,7 @@ def _check_requirement_satisfied(requirement_str):
                 requirement=requirement_str,
             )
 
-    if (
-        pkg_name == "mlflow"
-        and installed_version == mlflow.__version__
-        and Version(installed_version).is_devrelease
-    ):
+    if pkg_name == "mlflow" and Version(installed_version).is_devrelease:
```

Comment:
The intent here is "do not bother us with a version mismatch warning if mlflow version is dev". We don't need `installed_version == mlflow.__version__`.